src/coreclr/nativeaot/System.Private.CoreLib/src/System/Diagnostics/StackTrace.NativeAot.cs

```diff
@@ -95,5 +95,10 @@ internal void ToString(TraceFormat traceFormat, StringBuilder builder)
                 builder.AppendLine();
         }
 #endif
+
+        private StackFrame[] GetFramesCore()
+        {
+            return _stackFrames ?? [];
```

Comment:
Should NativeAOT get the same optimization? We do not want NativeAOT performance to be lagging behind.

Reply:
I'm not sure if debug info resolution is as significant on NAOT; I haven't benched it. My driving scenario is JIT-only for entirely separate reasons anyway as well. It shouldn't be too hard to do there, though I'd want to profile it first.

Reply:
#122227 is adding support for reading line information on the native AOT side so this optimization becomes relevant there too.